src/surfaceintersection.jl

```diff
@@ -0,0 +1,124 @@
+# Detect if the solution crossed a root of event function g
```

Comment:
Could this be used to define a Poincare-surface? Please, add docstrings.

Reply:
Yes, this can be used to define a Poincaré surface, but so far haven't found a way to do it automatically; it requires a bit of work from the user for each particular case; see also discussion in [this comment](https://github.com/PerezHz/TaylorIntegration.jl/pull/38#discussion_r139581165), above.

Sure, I will add docstrings! Thanks!

Reply:
I was thinking of maybe adding the Hénon-Heiles Poincaré map example as a notebook to the examples folder, what do you think?